reactor-core/src/main/java/reactor/core/publisher/FluxHandle.java

```diff
@@ -130,7 +148,14 @@ public boolean tryOnNext(T t) {
 				handler.accept(t, this);
 			}
 			catch (Throwable e) {
-				onError(Operators.onOperatorError(s, e, t, actual.currentContext()));
+				Throwable e_ = Operators.onNextError(t, e, actual.currentContext(), s);
+				if (e_ != null) {
+					onError(e_);
+				}
+				else {
+					reset();
+					s.request(1);
```

Comment:
shouldn't be needed in `tryOnNext`, the `return false` will be interpreted as "request more"

Reply:
this is onNext() though.

Reply:
This is onNext() though.

Reply:
nope it isn't 😄